mithril-aggregator/src/file_uploaders/gcp_uploader.rs

```diff
@@ -101,38 +251,253 @@ mod tests {
+            assert_eq!(expected_file_uri, file_uri);
+        }
+
+        #[tokio::test]
+        async fn upload_public_file_succeeds_with_overwriting_allowed() {
+            let allow_overwrite = true;
+            let local_file_path = Path::new("local_folder").join("snapshot.xxx.tar.gz");
+            let remote_folder_path = CloudRemotePath::new("remote_folder");
+            let remote_file_path = remote_folder_path.join("snapshot.xxx.tar.gz");
+            let expected_file_uri =
+                FileUri("https://cloud-host/remote_folder/snapshot.xxx.tar.gz".to_string());
+            let cloud_backend_uploader = {
+                let mut mock_cloud_backend_uploader = MockCloudBackendUploader::new();
+                let expected_file_uri_clone = expected_file_uri.clone();
+                mock_cloud_backend_uploader
+                    .expect_upload_file()
+                    .with(eq(local_file_path.clone()), eq(remote_file_path.clone()))
+                    .return_once(move |_, _| Ok(expected_file_uri_clone))
+                    .once();
+                mock_cloud_backend_uploader
+                    .expect_make_file_public()
+                    .with(eq(remote_file_path))
+                    .return_once(move |_| Ok(()))
+                    .once();
+
+                mock_cloud_backend_uploader
+            };
+            let file_uploader = GcpUploader::new(
+                Arc::new(cloud_backend_uploader),
+                remote_folder_path,
+                allow_overwrite,
+            );
+
+            let file_uri = file_uploader.upload(&local_file_path).await.unwrap();
+
+            assert_eq!(expected_file_uri, file_uri);
+        }
+
+        #[tokio::test]
+        async fn upload_public_file_fails_when_file_exists_fails_and_without_overwriting_allowed() {
+            let allow_overwrite = false;
+            let local_file_path = Path::new("local_folder").join("snapshot.xxx.tar.gz");
+            let remote_folder_path = CloudRemotePath::new("remote_folder");
+            let remote_file_path = remote_folder_path.join("snapshot.xxx.tar.gz");
+            let cloud_backend_uploader = {
+                let mut mock_cloud_backend_uploader = MockCloudBackendUploader::new();
+                mock_cloud_backend_uploader
+                    .expect_file_exists()
+                    .with(eq(remote_file_path))
+                    .return_once(move |_| Err(anyhow!("file exists error")))
+                    .once();
+
+                mock_cloud_backend_uploader
+            };
+            let file_uploader = GcpUploader::new(
+                Arc::new(cloud_backend_uploader),
+                remote_folder_path,
+                allow_overwrite,
+            );
+
+            file_uploader
+                .upload(&local_file_path)
+                .await
+                .expect_err("should have failed");
+        }
+
+        #[tokio::test]
+        async fn upload_public_file_fails_when_upload_fails() {
```

Comment:
Same simplifications can be done as for `upload_public_file_fails_when_file_exists_fails_and_without_overwriting_allowed`

Reply:
Done